handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -779,8 +780,13 @@ private void wrapNonAppData(ChannelHandlerContext ctx, boolean inUnwrap) throws
                     // If this is not enough we will increase the buffer in wrap(...).
                     out = allocateOutNetBuf(ctx, 2048, 1);
                 }
+
                 SSLEngineResult result = wrap(alloc, engine, Unpooled.EMPTY_BUFFER, out);
 
+                if (previousNeedUnwrap) {
```

Comment:
changes to SslHandler are just to support debugging and will be removed if this PR is accepted.